### ISSUE TITLE
DTU training time

### ISSUE THREAD

**Issue by alvin528** [hbb1/2d-gaussian-splatting#3]
Great work, congrats! 
I'm confused about the training time in DTU exp. In your paper, it costs about 5.5 minutes/15k iters on one 3090 GPU.  But I spent 30 minutes finishing it with the official command. Is there any modification needed for DTU data? 
Thanks again for your awesome work.

**Comment by hbb1** [comment]
Hi, I guess you use the full resolution? In our experiments, we use a downscaled resolution.

**Comment by Yubel426** [comment]
Hi, I'm having a similar problem and I've used the downscaled resolution. I'm testing on RTX4090 but it took an hour. The full logs are as follows:

python train.py -s ../../datasets/DTU_mask/scan63 -m output/scan63 -r 2 --depth_ratio 1
Optimizing output/scan63
Output folder: output/scan63 [06/05 11:46:11]
Tensorboard not available: not logging progress [06/05 11:46:11]
Reading camera 49/49 [06/05 11:46:11]
Loading Training Cameras [06/05 11:46:11]
Loading Test Cameras [06/05 11:46:27]
Number of points at initialisation :  17763 [06/05 11:46:27]
Training progress:  23%|█▊      | 7000/30000 [14:57<1:04:39,  5.93it/s, Loss=0.01721, distort=0.00018, normal=0.00000, Points=146921]
[ITER 7000] Evaluating train: L1 0.009540539048612118 PSNR 33.23699951171875 [06/05 12:01:26]

[ITER 7000] Saving Gaussians [06/05 12:01:26]
Training progress: 100%|███████| 30000/30000 [1:10:44<00:00,  7.07it/s, Loss=0.00901, distort=0.00008, normal=0.00480, Points=171687]

[ITER 30000] Evaluating train: L1 0.006552549358457328 PSNR 37.226990127563475 [06/05 12:57:13]

[ITER 30000] Saving Gaussians [06/05 12:57:13]

Training complete. [06/05 12:57:15]

**Comment by Yubel426** [comment]
When I assigned more cpu cores to this task, the training time became normal.
I noticed that 2dgs seems to require an unusually large number of cpu cores, at least when I tested it, it required 30 cores, whereas I only needed 1 core when training 3dgs.
I will continue to test this.

The log of `top` is as follows
    PID USER      PR  NI    VIRT    RES    SHR S  %CPU  %MEM     TIME+ COMMAND
4128407 xxx       20   0   45.0g   1.1g 333224 S  **3016**  0.2   8:29.17 python

**Comment by hbb1** [comment]
I am not sure if it is related to this line of code https://github.com/hbb1/2d-gaussian-splatting/blob/main/utils/point_utils.py#L20C1-L21C106. 

```python
    # TODO: reduce dynamic allocation for speed 
    points = torch.stack([grid_x, grid_y, torch.ones_like(grid_x)], dim=-1).reshape(-1, 3).float().cuda()
```

If yes, one possible solution is to pre-store the points (ray direction) in the Camera class. Please let me know if this helps.

**Comment by Yubel426** [comment]
Thanks!
When I commented out the line https://github.com/hbb1/2d-gaussian-splatting/blob/5c018d5c45fbdadd4170a932ae8f6fab7c5666cb/gaussian_renderer/__init__.py#L138 the CPU usage returned to normal.
`surf_point` seems to be a useless variable? 
Maybe you can refer to https://github.com/Asparagus15/GaussianShader/blob/de77861c0ee29de6ad797b5ad16d8d3e35e59e76/utils/graphics_utils.py#L122

**Comment by hbb1** [comment]
@Yubel426  Looks good. But do you mean simply removing surf_point resolves the problem for you?

**Comment by Yubel426** [comment]
I have simply resolved the problem by moving this operation to the GPU, reducing CPU usage. Although there was a slight increase in VRAM usage, the speed has also seen some improvement.
I have submitted a PR  #8 . You can test it!

**Comment by hbb1** [comment]
Thank you! I will test it.

**Comment by alvin528** [comment]
> I have simply resolved the problem by moving this operation to the GPU, reducing CPU usage. Although there was a slight increase in VRAM usage, the speed has also seen some improvement. I have submitted a PR #8 . You can test it!

It seems good.  Thank you
